### ISSUE TITLE
List of mapalgo options for streams is different for MCT than NUOPC, currently we are using the MCT list

### ISSUE THREAD

**Issue by ekluzek** [ESCOMP/CTSM#1912]
### Brief summary of bug

The list of options for the mapalgo namelist fields in CTSM is the list for MCT, rather than NUOPC. Since, we are moving to NUOPC we should update it to the NUOPC list and remove the MCT only options.

### General bug information

**CTSM version you are using:** ctsm5.1.dev114 (but comes with the introduction of NUOPC)

**Does this bug cause significantly incorrect results in the model's science?** No
**Configurations affected:** with NUOPC

### Details of bug

The list of mapalgo streams options in the namelist are:

ndepmapalgo
lai_mapalgo
lightngmapalgo
popdensmapalgo
urbantvmapalgo

The list of stream options for mapalgo for MCT is:

```
    bilinear = bilinear interpolation
    nn       = nearest neighbor
    nnoni    = nearest neighbor on the "i" (longitude) axis
    nnonj    = nearest neighbor on the "j" (latitude) axis
    spval    = set to special value
    copy     = copy using the same indices
```

While the list for NUOPC is:

```
    bilinear = bilinear interpolation
    nn       = nearest neighbor
    consf  = first order conservative mapping
    consd = second order conservative mapping?
    none    = don't do any interpolation (same as copy for MCT?)
```

**Comment by mvertens** [comment]
@ekluzek - nuopc does not have nnoni, nnonj, spval or copy. These were things specific to the cpl7 data models.
I am also in the process of refactoring how stream namelists in CTSM are specified (by using xml variables directly).
This will enable us to more easily remove streams (like ndep). I am in the process of testing this and will issue a PR hopefully shortly.

**Comment by ekluzek** [comment]
@mvertens yes that's the point. As setup now when using NUOPC, you could set

lightngmapalgo=spval

and build-namelist will allow you to do that, but it'll fail at run-time with an error.

Conversely, you could set

lightngmapalgo=consf

which is a valid option for NUOPC, but as it is, it will fail with an error at build-namelist time as consf isn't in the list of valid options for lightngmapalgo.

**Comment by billsacks** [comment]
May be addressed by #1929

**Comment by ekluzek** [comment]
The list changed slightly to: bilinear, nn, redist, mapconsd, mapconf, none
